### ISSUE TITLE
rain deploy fails with "SerializationError: failed to decode query XML error response"

### ISSUE THREAD

**Issue by ikben** [aws-cloudformation/rain#33]
Rain version:
```
~# ~/go/bin/rain --version
Rain v0.9.0 darwin/amd64
```

This works:
```
~# aws cloudformation deploy \
  --stack-name landing-zone \
  --template-file output/custom-landing-zone-initiation.template \
  --s3-bucket cf-templates-XXXXX-eu-west-1 \
  --capabilities CAPABILITY_NAMED_IAM

Waiting for changeset to be created..
Waiting for stack create/update to complete
Successfully created/updated stack - landing-zone
```

However, this fails:

```
~/go/bin/rain deploy output/custom-landing-zone-initiation.template landing-zone -f --debug
DEBUG: Loading AWS config
DEBUG: Trying default configs...
DEBUG: ...and they're valid
Deploying 'custom-landing-zone-initiation.template' as 'landing-zone' in eu-west-1:
Preparing template... DEBUG: Artifact bucket: rain-artifacts-123456789012-eu-west-1
DEBUG: Package output:
Successfully packaged artifacts and wrote output template to file /var/folders/pq/cvfgj3_s0c565_sc9rwywd0w0000gn/T/768452183.
Execute the following command to deploy the packaged template
aws cloudformation deploy --template-file /var/folders/pq/cvfgj3_s0c565_sc9rwywd0w0000gn/T/768452183 --stack-name <YOUR STACK NAME>

DEBUG: Loading packaed template file
Checking current status of stack 'landing-zone'... DEBUG: Stack exists
DEBUG: Handling parameters
DEBUG: Parameters: [{
  ParameterKey: "SubscribeAllChangeEventsEmailToTopic",
  UsePreviousValue: true
} {
  ParameterKey: "PipelineApprovalStage",
  UsePreviousValue: true
} {
  ParameterKey: "NotificationAddonsEmailParameter",
  UsePreviousValue: true
} {
  ParameterKey: "KmsAdminRoleName",
  UsePreviousValue: true
} {
  ParameterKey: "CodecommitAccountId",
  UsePreviousValue: true
} {
  ParameterKey: "SecurityAlertEmail",
  UsePreviousValue: true
} {
  ParameterKey: "PipelineApprovalEmail",
  UsePreviousValue: true
} {
  ParameterKey: "AddonsManagedParameter",
  UsePreviousValue: true
} {
  ParameterKey: "CodecommitArtifactBucket",
  UsePreviousValue: true
} {
  ParameterKey: "LandingZoneBaselinePortfolioId",
  UsePreviousValue: true
} {
  ParameterKey: "AllChangeEventsEmail",
  UsePreviousValue: true
}]
DEBUG: Creating change set
Creating change set ·˙.DEBUG: Removing temporary template file: /var/folders/pq/cvfgj3_s0c565_sc9rwywd0w0000gn/T/768452183
panic: Error creating changeset: SerializationError: failed to decode query XML error response
caused by: EOF [recovered]
	panic: Error creating changeset: SerializationError: failed to decode query XML error response
caused by: EOF

goroutine 1 [running]:
github.com/aws-cloudformation/rain/cmd.Execute.func1()
	/Users/ben/go/pkg/mod/github.com/aws-cloudformation/rain@v0.9.0/cmd/wrap.go:76 +0x19d
panic(0x1c5d800, 0xc0016701b0)
	/usr/local/Cellar/go/1.14.3/libexec/src/runtime/panic.go:969 +0x166
github.com/aws-cloudformation/rain/cmd.glob..func5(0x3bedc40, 0xc0006c4880, 0x2, 0x4)
	/Users/ben/go/pkg/mod/github.com/aws-cloudformation/rain@v0.9.0/cmd/deploy.go:281 +0x203c
github.com/spf13/cobra.(*Command).execute(0x3bedc40, 0xc0006c4840, 0x4, 0x4, 0x3bedc40, 0xc0006c4840)
	/Users/ben/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:846 +0x29d
github.com/spf13/cobra.(*Command).ExecuteC(0x3bed9a0, 0xc00069ff78, 0x100772f, 0xc000110058)
	/Users/ben/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/ben/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
github.com/aws-cloudformation/rain/cmd.Execute(0x3bed9a0)
	/Users/ben/go/pkg/mod/github.com/aws-cloudformation/rain@v0.9.0/cmd/wrap.go:84 +0x4d
main.main()
	/Users/ben/go/pkg/mod/github.com/aws-cloudformation/rain@v0.9.0/main.go:22 +0x2d
```

The changeset does take a while to generate in the console.

**Comment by stilvoid** [comment]
I've updated the version of the AWS SDK used in rain in v0.10.0 and also improved error reporting. Would you be able to supply me a copy of the template that didn't work or try it again with the latest version of rain? Thanks for any further info.

**Comment by ikben** [comment]
This is what the latest version of rain gives me:
```
 ./rain deploy output/custom-landing-zone-initiation.template landing-zone -f --debug
DEBUG: Loading AWS config
Deploying 'custom-landing-zone-initiation.template' as 'landing-zone' in eu-west-1:
DEBUG: Preparing template
Preparing template ·˙.DEBUG: Artifact bucket: rain-artifacts-123456789012-eu-west-1
Preparing template .· DEBUG: Package output:
Successfully packaged artifacts and wrote output template to file /var/folders/pq/cvfgj3_s0c565_sc9rwywd0w0000gn/T/790128142.
Execute the following command to deploy the packaged template
aws cloudformation deploy --template-file /var/folders/pq/cvfgj3_s0c565_sc9rwywd0w0000gn/T/790128142 --stack-name <YOUR STACK NAME>

DEBUG: Loading packaged template file
DEBUG: Checking current status of stack 'landing-zone'...
Checking current status of stack 'landing-zone'...   .˙DEBUG: Stack exists
DEBUG: Handling parameters
DEBUG: Removing temporary template file: /var/folders/pq/cvfgj3_s0c565_sc9rwywd0w0000gn/T/790128142
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x227be96]

goroutine 1 [running]:
github.com/aws-cloudformation/rain/cmd.Execute.func1()
	/home/steve/code/github.com/aws-cloudformation/rain/cmd/wrap.go:76 +0x19d
panic(0x2398e60, 0x3022900)
	/usr/lib/go/src/runtime/panic.go:969 +0x175
github.com/aws-cloudformation/rain/cmd.glob..func5(0x302c100, 0xc00004d140, 0x2, 0x4)
	/home/steve/code/github.com/aws-cloudformation/rain/cmd/deploy.go:285 +0x8b6
github.com/spf13/cobra.(*Command).execute(0x302c100, 0xc00004d100, 0x4, 0x4, 0x302c100, 0xc00004d100)
	/home/steve/.gohome/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:854 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0x302be60, 0xc00012df78, 0x1005ea5, 0xc00010a058)
	/home/steve/.gohome/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:958 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/home/steve/.gohome/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:895
github.com/aws-cloudformation/rain/cmd.Execute(0x302be60)
	/home/steve/code/github.com/aws-cloudformation/rain/cmd/wrap.go:84 +0x4d
main.main()
	/home/steve/code/github.com/aws-cloudformation/rain/main.go:22 +0x2d
```

**Comment by stilvoid** [comment]
Ok, I've figured out what this is. It's that the template is too large to supply as a `TemplateBody` to `CreateChangeSet`. `aws cloudformation deploy` handles this for you by dropping the template into a bucket first. I'll do the same in rain :)

Aiming to get that fix in place this weel :+1:

**Comment by stilvoid** [comment]
Ok this is fixed in 069ae100ffcb393fbe716ca06c716fd8fbf6f5da. A few more things to do but will likely release as v0.10.1 later today.
